pycqed/measurement/sweep_points.py

```diff
@@ -44,26 +44,63 @@ class SweepPoints(list):
+    def __init__(self, param=None, values=None, unit='', label=None,
+                 dimension=-1, min_length=0):
+        """
+        Create a SweepPoints instance.
+        :param param: list of dicts, repr of SweepPoints instance, or name of
+            a sweep parameter
+        :param values: list or array of numeric values of the sweep parameter
+            specified by param
+        :param unit: string specifying the unit of the sweep parameter
+            specified by param
+        :param label: str specifying the (latex style) label/name of the sweep
+            parameter specified by param
+        :param dimension: sweep dimension where sweep parameter param should be
+            added
+        :param min_length: minimum number of sweep dimensions to create
+        """
         super().__init__()
-        if param_name is not None and values is not None:
-            self.add_sweep_parameter(param_name, values, unit, label,
-                                     dimension)
-        elif from_dict_list is not None:
-            for d in deepcopy(from_dict_list):
-                if len(d) == 0 or isinstance(list(d.values())[0], tuple):
-                    # assume that dicts have the same format as this class
-                    self.append(d)
-                else:
-                    # import from a list of sweep dicts in the old format
-                    self.append({k: (v['values'],
-                                     v.get('unit',''),
-                                     v.get('label', k))
-                                 for k, v in d.items()})
+        if isinstance(param, list):
+            self.add_dict_list(param)
+        elif isinstance(param, str):
+            try:
+                self.add_dict_list(eval(param))
+            except NameError:
+                if values is not None:
+                    self.add_sweep_parameter(param, values, unit, label,
+                                             dimension)
+
         while len(self) < min_length:
             self.add_sweep_dimension()
 
+    def __getitem__(self, i):
+        """
+        Overloading of List.__getitem__ to ensure type SweepPoints is preserved.
+        :param i: element or slice
+        :return: element or new SweepPoints instance
+        """
+        new_data = super().__getitem__(i)
+        if type(i) == slice:
+            new_data = self.__class__(new_data)
+        return new_data
+
+    def add_dict_list(self, dict_list):
+        """
+        Append the dicts in dict_list to self.
+        :param dict_list: list of dictionaries in the format of this class
+        """
+        for d in deepcopy(dict_list):
+            if len(d) == 0 or isinstance(list(d.values())[0], tuple):
+                # assume that dicts have the same format as this class
+                self.append(d)
+            else:
+                # import from a list of sweep dicts in the old format
```

Comment:
the docstring should mention this backward-compatibility

Reply:
👍 Done now